src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -343,6 +376,9 @@ internal override int CompareAnyKeys(int index1, int index2)
         protected override void QuickSort(int[] keys, int lo, int hi) =>
             new Span<int>(keys, lo, hi - lo + 1).Sort(CompareAnyKeys);
 
+        protected override void QuickSortElements(TElement[] element, int lo, int hi) =>
+            new Span<TKey>(_keys, lo, hi - lo + 1).Sort(new Span<TElement>(element, lo, hi-lo +1), _comparer);
```

Comment:
I think this needs some check on the bounds of the span be allocated. The stack isn't unlimited in size and can differ by platform.

Reply:
Can you expand on this? My understanding is that element and _keys are both heap allocated and the only stack allocation in the Span is the pointer to the first entry in the arrays and the length.

Reply:
You're right, reading comprehension failure on my part.

Reply:
Is there a reason you use new Span, instead of the more common AsSpan?

Reply:
I was just following convention of the previous method. I can switch both methods to AsSpan if that is the common pattern.